tests/models/test_pyro.py

```diff
@@ -156,9 +207,19 @@ def test_pyro_bayesian_regression(save_path):
 def test_pyro_bayesian_regression_jit():
     use_gpu = int(torch.cuda.is_available())
     adata = synthetic_iid()
+    # add index for each cell (provided to pyro plate for correct minibatching)
+    adata.obs["_indices"] = np.arange(adata.n_obs).astype("int64")
+    register_tensor_from_anndata(
+        adata,
+        registry_key="ind_x",
+        adata_attr_name="obs",
+        adata_key_name="_indices",
+    )
     train_dl = AnnDataLoader(adata, shuffle=True, batch_size=128)
     pyro.clear_param_store()
-    model = BayesianRegressionModule(adata.shape[1], 1)
+    model = BayesianRegressionModule(
+        in_features=adata.shape[1], out_features=1, n_obs=adata.n_obs
```

Comment:
the training plan actually sets n_obs automatically (see https://github.com/YosefLab/scvi-tools/blob/6dae6482efa2d235182bf4ad10dbd9483b7d57cd/scvi/train/_trainingplans.py#L602-L606)